### DISCUSSION TITLE
Rendering Rich objects in Prompt-toolkit widgets

### DISCUSSION THREAD

**Discussion by byt3bl33d3r** [Textualize/rich#936] [Q&A] [answered]
Heya,

I'm trying to render Rich tables in a full screen prompt-toolkit application in the `TextArea` widget. Is there a way turning Rich objects into a string , or some other way that is recommended to do this? 

Thanks

**Comment by willmcgugan** [comment, marked as answer]
Yep https://rich.readthedocs.io/en/stable/console.html

**Reply by willmcgugan** [reply]
See the Capturing Output section...

**Reply by byt3bl33d3r** [reply]
I completely missed that... (and I don't know how). thanks again and sorry for the bother.

**Reply by lllama** [reply]
@byt3bl33d3r I don't suppose you have any examples for wiring this up? I'm interested in using prompt-toolkit to add some key-bindings to my Rich dashboard.

**Comment by brestows** [comment]
Hi, I have code:
```#!/usr/bin/env python

from prompt_toolkit.application import Application
from prompt_toolkit.layout.containers import HSplit
from prompt_toolkit.layout.layout import Layout
from prompt_toolkit.widgets import TextArea
from prompt_toolkit.widgets.base import HorizontalLine


from io import StringIO
from rich.console import Console

console = Console(file=StringIO(), highlight=True, force_terminal=True, color_system="truecolor")

def main():

    hl = HorizontalLine()
    output_field = TextArea()
    area = TextArea(height=1, prompt=">>> ", style="class:input-field", multiline=False, wrap_lines=False, )
    input_field = area
    container = HSplit(
        [
            output_field,
            hl,
            input_field,
        ]
    )

    console.print("[bold]Hello!")
    txt = console.file.getvalue()
    output_field.text = txt

    application = Application(
        layout=Layout(container, focused_element=input_field),
        full_screen=True,
    )
    application.run()


if __name__ == "__main__":
    main()

```
  why is it  rendering rich objects  not correctly ?
![image](https://user-images.githubusercontent.com/827907/202816883-f47eea48-d9b5-4822-a58a-3f4552a3f633.png)

**Reply by thvitt** [reply]
Try:

```python
from prompt_toolkit.formatted_text import ANSI, to_formatted_text
txt = to_formatted_text(ANSI(console.file.getvalue())
```
The ANSI text that rich writes to the console or capture string needs be converted to prompt toolkit’s implementation of formatted strings first, it doesn’t just pass through the ANSI sequences.

**Reply by AtticusZeller** [reply]
it works!!!
